library/src/main/java/com/novoda/downloadmanager/lib/DownloadReceiver.java

```diff
@@ -142,7 +148,7 @@ private void sendNotificationClickedIntent(Context context, long[] ids) {
      * {@link DownloadManager#COLUMN_ID}.
      */
     private void openDownload(Context context, long id) {
-        Intent intent = OpenHelper.buildViewIntent(context, id);
+        Intent intent = new OpenHelper(downloadsUriProvider).buildViewIntent(context, id);
```

Comment:
did you consider the impact of this change, creating an object where there was not one before